### PR TITLE
Added #4897: Add table_privileges, schema_privileges and user_privile…

### LINKED ISSUE #4897: Add table_privileges, schema_privileges and user_privileges table in information_schema database.
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by luozenglin**
…ges table in information_schema database.

## Proposed changes
Add table_privileges, schema_privileges and user_privileges table in information_schema database.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on (Fix #4897 ), and have described the bug/feature there in detail
- [] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [] If this change need a document change, I have updated the document
- [] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by EmmyMiao87** [comment]
Could you please add some description about this pr？

File: fe/fe-core/src/main/java/org/apache/doris/catalog/SchemaTable.java [hunk 1]
```diff
@@ -104,11 +104,32 @@ public static Builder builder() {
...
+                                    .build()))
+                    .put("schema_privileges", new SchemaTable(
+                            SystemIdGenerator.getNextId(),
+                            "schema_privileges",
+                            TableType.SCHEMA,
+                            builder()
+                                    .column("GRANTEE", ScalarType.createVarchar(NAME_CHAR_LEN))
+                                    .column("TABLE_CATALOG", ScalarType.createVarchar(FN_REFLEN))
+                                    .column("TABLE_SCHEMA", ScalarType.createVarchar(NAME_CHAR_LEN))
+                                    .column("PRIVILEGE_TYPE", ScalarType.createVarchar(64))
```
[the remark this answers is not in the document]
Comment (luozenglin) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/SchemaTable.java:121]: Good idea.

File: fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/PaloAuth.java [hunk 2]
```diff
@@ -1300,6 +1301,120 @@ public TFetchResourceResult toResourceThrift() {
...
+                String tblName = tblPrivEntry.getOrigTbl();
+
+                if (dbName.equals("information_schema" /* Don't show privileges in information_schema */)
+                        || !checkTblPriv(currentUser, tblPrivEntry.getOrigDb(), tblName, PrivPredicate.SHOW)) {
+                    continue;
+                }
+
+                String grantee = new String("\'").concat(ClusterNamespace.getNameFromFullName(tblPrivEntry.getOrigUser()))
+                        .concat("\'@\'").concat(tblPrivEntry.getOrigHost()).concat("\'");
+                String isGrantable = tblPrivEntry.getPrivSet().get(2) ? "yes" : "no"; //Grant_priv
```
[the remark this answers is not in the document]
Comment (luozenglin) [reply fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/PaloAuth.java:1324]: Should use uppercase, I will modify it.

File: be/src/exec/schema_scanner/schema_user_privileges_scanner.cpp [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    }
+    RETURN_IF_ERROR(get_new_table());
+    return Status::OK();
+}
+
+Status SchemaUserPrivilegesScanner::fill_one_row(Tuple *tuple, MemPool *pool) {
+    // set all bit to not null
+    memset((void *)tuple, 0, _tuple_desc->num_null_bytes());
+    const TPrivilegeStatus& tbl_priv_status = _priv_result.privileges[_priv_index];
+    // grantee
```
Comment (morningman) [inline be/src/exec/schema_scanner/schema_user_privileges_scanner.cpp:56]:
```
    {
        void *slot = tuple->get_slot(_tuple_desc->slots()[0]->tuple_offset());
        StringValue* str_slot = reinterpret_cast<StringValue*>(slot);
        const std::string* src = &tbl_priv_status.grantee;
        str_slot->len = src->length();
        str_slot->ptr = (char *)pool->allocate(str_slot->len);
        if (NULL == str_slot->ptr) {
            return Status::InternalError("Allocate memcpy failed.");
        }
        memcpy(str_slot->ptr, src->c_str(), str_slot->len);
    }
```

This kind of code block is used everywhere, I think we can extract a common method for it.

**Review by EmmyMiao87 — COMMENTED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**